Packs/ZoomMail/Integrations/ZoomMail/ZoomMail.py

```diff
@@ -417,32 +428,31 @@ def fetch_incidents(client: ZoomMailClient, params: dict[str, str]) -> None:
 
         # Check if it's a new message, and if it's either all messages (fetch_threads is False) or only thread starters
         if (internal_date > last_fetch_info["internalDate"] or (
-            internal_date == last_fetch_info["internalDate"] and message_id not in last_fetched_ids
+            internal_date == last_fetch_info["internalDate"] and message_id not in last_fetch_info["ids"]
         )) and (fetch_threads or (not fetch_threads and message_id == thread_id)):
             incident = zoom_mail_to_incident(message_details, client, fetch_from)
             incidents.append(incident)
-            message_dates.append(internal_date)
+            if not last_page_fetch_token:
+                message_dates.append(internal_date)
 
-    if message_dates:
+    if message_dates and not next_page_token:
```

Comment:
if multiple pages has messages of the same time, we need to save ids of the latest time from multiple pages, if time is inclusive.

this is a delicate case. I think you may need to save the local max time between pages. lets talk about this.